File: app/src/main/java/io/github/marcocipriani01/telescopetouch/activities/SplashScreenModule.java
package io.github.marcocipriani01.telescopetouch.activities;

import android.app.Activity;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import androidx.fragment.app.FragmentManager;

import dagger.Module;
import dagger.Provides;
import io.github.marcocipriani01.telescopetouch.R;
import io.github.marcocipriani01.telescopetouch.activities.dialogs.EulaDialogFragment;
import io.github.marcocipriani01.telescopetouch.inject.PerActivity;

/**
 * Created by johntaylor on 4/2/16.
 */
@Module
public class SplashScreenModule {

    private final SplashScreenActivity activity;

    public SplashScreenModule(SplashScreenActivity activity) {
        this.activity = activity;
    }

    @Provides
    @PerActivity
    EulaDialogFragment provideEulaDialogFragment() {
        EulaDialogFragment fragment = new EulaDialogFragment();
        fragment.setEulaAcceptanceListener(activity);
        return fragment;
    }

    @Provides
    @PerActivity
    Activity provideActivity() {
        return activity;
    }

    @Provides
    @PerActivity
    Animation provideFadeoutAnimation() {
        return AnimationUtils.loadAnimation(activity, R.anim.fadeout);
    }

    @Provides
    @PerActivity
    FragmentManager provideFragmentManager() {
        return activity.getSupportFragmentManager();
    }
}

File: app/src/main/java/com/google/android/stardroid/activities/SplashScreenModule.kt
package com.google.android.stardroid.activities

import android.app.Activity
import android.app.FragmentManager
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import com.google.android.stardroid.R
import com.google.android.stardroid.activities.dialogs.EulaDialogFragment
import com.google.android.stardroid.activities.dialogs.WhatsNewDialogFragment
import com.google.android.stardroid.inject.PerActivity
import dagger.Module
import dagger.Provides

/**
 * Created by johntaylor on 4/2/16.
 */
@Module
class SplashScreenModule(private val activity: SplashScreenActivity) {
    @Provides
    @PerActivity
    fun provideEulaDialogFragment(): EulaDialogFragment {
        val fragment = EulaDialogFragment()
        fragment.setEulaAcceptanceListener(activity)
        return fragment
    }

    @Provides
    @PerActivity
    fun provideWhatsNewDialogFragment(): WhatsNewDialogFragment {
        val whatsNewDialogFragment = WhatsNewDialogFragment()
        whatsNewDialogFragment.setCloseListener(activity)
        return whatsNewDialogFragment
    }

    @Provides
    @PerActivity
    fun provideActivity(): Activity {
        return activity
    }

    @Provides
    @PerActivity
    fun provideFadeoutAnimation(): Animation {
        return AnimationUtils.loadAnimation(activity, R.anim.fadeout)
    }

    @Provides
    @PerActivity
    fun provideFragmentManager(): FragmentManager {
        return activity.getFragmentManager()
    }
}